core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -325,4 +324,10 @@ private void deleteDataFiles(Set<String> manifestsToScan, Set<String> manifestsT
       return CloseableIterable.withNoopClose(snapshot.manifests());
     }
   }
+
+  private ManifestFile pruneManifest(ManifestFile manifest) {
```

Comment:
Do we expect the partition summary to be very big? I think this is safe since it is just looking for files to delete, but I'm not sure that it is necessary to discard stats.

Reply:
Not sure either. Previously, we kept only file names, so I wanted to use as less memory as possible, but I don't think the partition summary will be too big. Maybe, we can keep it for now and add pruning later if needed? What do you think, @rdblue?

Reply:
I removed truncation for now.